modules/basicfuncs/ip-funcs.c

```diff
@@ -40,3 +42,121 @@ tf_ipv4_to_int(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_ipv4_to_int);
+
+typedef struct _DnsResolveIpState
+{
+  TFSimpleFuncState super;
+  HostResolveOptions host_resolve_options;
+} DnsResolveIpState;
+
+static gboolean
+_process_use_fqdn(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_fqdn = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_use_dns(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_dns = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_dns_cache(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_dns_cache = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_normalize_hostnames(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->normalize_hostnames = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_parse_host_resolve_options(gint *argc, gchar **argv[], HostResolveOptions *host_resolve_options, GError **error)
+{
+  gboolean result = FALSE;
+  GOptionEntry entries[] =
+  {
+    { "use-fqdn", 'f', 0, G_OPTION_ARG_CALLBACK, _process_use_fqdn, NULL, NULL },
+    { "use-dns", 'd', 0, G_OPTION_ARG_CALLBACK, _process_use_dns, NULL, NULL },
+    { "dns-cache", 'c', 0, G_OPTION_ARG_CALLBACK, _process_dns_cache, NULL, NULL },
+    { "normalize-hostnames", 'n', 0, G_OPTION_ARG_CALLBACK, _process_normalize_hostnames, NULL, NULL },
+    { NULL }
+  };
+
+  GOptionContext *ctx = g_option_context_new(*argv[0]);
+  GOptionGroup *group = g_option_group_new("host-resolve-options", NULL, NULL, host_resolve_options, NULL);
+  g_option_group_add_entries(group, entries);
+  g_option_context_set_main_group(ctx, group);
+
+  if (!g_option_context_parse(ctx, argc, argv, error))
+    goto exit;
+
+  result = TRUE;
+exit:
+  g_option_context_free(ctx);
+  return result;
+}
+
+static gboolean
+_setup_host_resolve_options(DnsResolveIpState *state, gint *argc, gchar **argv[], GError **error)
+{
+  host_resolve_options_global_defaults(&state->host_resolve_options);
```

Comment:
Have you considered defaulting to the values in global options ? Right now the default values are the default value of global options, and not the value of global options (as normally it works for many options that has global counterpart).

Also I think it would be beneficial even if the current version is kept, you do not state with exact values what is the default (yes, no) but say that the same as the default values of the global options counterpart (as they are subject to change at any time).

Reply:
Actually, following the global options makes sense, yes. I believe we can access it from `..._prepare()`, where `LogTemplate *parent` is given as an argument, which has a `GlobalConfig *cfg` field.